swap/htlc.go

```diff
@@ -169,7 +196,57 @@ func NewHtlc(version ScriptVersion, cltvExpiry int32,
 	}, nil
 }
 
-// SwapHTLCScriptV1 returns the on-chain HTLC witness script.
+// GenSuccessWitness returns the success script to spend this htlc with
+// the preimage.
+func (h *Htlc) GenSuccessWitness(receiverSig []byte,
+	preimage lntypes.Preimage) (wire.TxWitness, error) {
+
+	if h.Hash != preimage.Hash() {
+		return nil, errors.New("preimage doesn't match hash")
+	}
+
+	return h.genSuccessWitness(receiverSig, preimage), nil
+}
+
+func (h *Htlc) GenTimeoutWitness(senderKey [33]byte,
+	senderSig []byte) wire.TxWitness {
+
+	return h.genTimeoutWitness(senderKey, senderSig)
```

Comment:
I think we can just unexport `genTimeoutWitness`?

Reply:
done